tst/Carnival/Unit/Hooks/HookSystemTest.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace Tests\Unit\Hooks;
+
+use Carnival\Hooks\Hook;
+use Tests\Carnival\CarnivalTest;
+use Tests\Carnival\Unit\Data\Hooks\ExampleHook;
+
+class HookSystemTest extends CarnivalTest {
+
+    protected $hookSystem;
+    protected $exampleHook;
+
+    public function setUp () : void {
+        parent::setUp();
+
+        $this->hookSystem = new Hook;
+        $this->exampleHook = new ExampleHook("Hello World!");
+    }
+
+    public function testCanAddHookToHookList () {
+        $this->addExampleHookToHookList();
+        $this->assertEquals(1, $this->hookSystem->count(ExampleHook::class));
+    }
+
+    public function testCanRemoveHookFromHookList () {
+        $this->addExampleHookToHookList();
+        $this->assertEquals(1, $this->hookSystem->count(ExampleHook::class));
+        
+        $this->hookSystem->remove(ExampleHook::class);
+        $this->assertEquals(0, $this->hookSystem->count(ExampleHook::class));
+    }
+
+    public function testCanExecuteHooksInHookList () {
+        $this->addExampleHookToHookList();
+        $this->assertEquals(1, $this->hookSystem->count(ExampleHook::class));
+
+        $this->assertEquals("Hello World!", $this->exampleHook->getText());
+
+        $this->hookSystem->execute($this->exampleHook);
+
+        $this->assertEquals("Goodbye World!", $this->exampleHook->getText());
+    }
+
+    public function testCanExecuteMultipleHooksForTheSameEvent () {
+        $this->addMultipleHooksToHookList();
+        $this->assertEquals(2, $this->hookSystem->count(ExampleHook::class));
+
+        $this->assertEquals("Hello World!", $this->exampleHook->getText());
+
+        $this->hookSystem->execute($this->exampleHook);
+
+        $this->assertEquals("Goodnight World!", $this->exampleHook->getText());
```

Comment:
This test doesn't really prove that both hooks ran, does it? Since the second overrides the first, it's possible for only the second hook to have run and this test would still pass?

Reply:
That is a good point, I'll look into a way to rework this and check that both fire somehow.

Reply:
Perhaps instead of modifying a string, they modify a number? One hook could increment the num by 1, and the other would increment it by two. So when both run, the num is incremented by three

Reply:
I like this idea. I will reimplement the tests around this concept.